capstone/src/components/User/Experience/RegisterExperience/RegisterExperience.jsx

```diff
@@ -11,62 +13,79 @@ function validateEmail(email) {
+  const { username } = useContext(UserContext);
+  // const [experienceValues, setExperienceValues] = useState(experienceData);
   const [error, setError] = useState({
     name: '',
-    location: '',
+    address: '',
     email: '',
-    description: '',
   });
+
   const fields = [
     {
-      id: 'name', value: experienceValues.name, displayText: 'Name', placeholder: 'Your amazing business', isRequired: true, type: 'text', input: 'input',
+      id: 'name', value: experienceValues.name, displayText: 'Name', placeholder: 'Your amazing business', isRequired: true, type: 'text', error: error.name,
     },
     {
-      id: 'location', value: experienceValues.location, displayText: 'Business Location', placeholder: '123 Stewart Drive', isRequired: true, type: 'text', input: 'input',
+      id: 'address', value: experienceValues.address, displayText: 'Business Address', placeholder: '123 Stewart Drive', isRequired: true, type: 'address', error: error.address,
     },
     {
-      id: 'email', value: experienceValues.email, displayText: 'Email', placeholder: 'yourbusiness@ext.org', isRequired: true, type: 'email', input: 'input',
+      id: 'email', value: experienceValues.email, displayText: 'Email', placeholder: 'yourbusiness@ext.org', isRequired: true, type: 'text', error: error.email,
     },
     {
-      id: 'description', value: experienceValues.description, displayText: 'Description', placeholder: 'Let people know your value in here!', isRequired: false, type: 'text', input: 'textArea',
+      id: 'description', value: experienceValues.description, displayText: 'Description', placeholder: 'Let people know your value in here!', isRequired: false, type: 'textArea', error: error.description,
     },
   ];
-  function handleRegistryChange(inputName, value) {
-    setExperienceValues({ ...experienceValues, [inputName]: value });
+  function onRegistryChange(inputName, value) {
+    setExperienceValues((experienceValues) => ({ ...experienceValues, [inputName]: value }));
+    setError({ ...error, [inputName]: '' });
   }
-  function handleRegistrySubmission(form) {
+
+  // used to display success message in the following function
+  const toast = useToast();
+  function onRegistrySubmission(form) {
+    // Check for errors
     let hasError = false;
     Object.keys(form).map((key) => {
       if (key === 'email') {
         if (!(validateEmail(form[key]))) {
           setError({ ...error, [key]: 'Not a valid email' });
           hasError = true;
         }
-      } else if (!(form[key])) {
-        setError({ ...error, [key]: 'Not a valid value' });
+      } else if (key === 'lat' || key === 'lng') {
+        if (form[key] === 0) {
+          setError({ ...error, address: 'Not a valid address' });
+          hasError = true;
+        }
+      } else if (!(form[key]) && key !== 'description') {
+        setError({ ...error, [key]: `Not a valid ${key}` });
         hasError = true;
       }
       return hasError;
     });
```

Comment:
nit: Although this looks like it works, it seems like it might not be the easiest to follow/read/understand in the future. I'll mention back to this comment https://github.com/carlosvegap/capstone-kickoff/pull/4#discussion_r918369797, using an object validation library might simplify things if you have time. *(low priority)*

Reply:
Thanks a lot! I created an issue about that to keep it in mind